-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7a2483/45027/summary.html
**COMMIT**: ce791a67089c4358d740c4c68258f0b5a7d449de
**CMSSW**: CMSSW_15_1_X_2025-03-18-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-externals/geant4/90/45027/install.sh` to create a dev area with all the needed externals and cmssw changes.


## External Build

I found compilation error when building: 

<pre>make[1]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/geant4/11.2.2-2f8e5d3e76b0f6b6f530aab595a96164/build'
make[1]: *** [CMakeFiles/Makefile2:1925: source/CMakeFiles/G4processes-static.dir/all] Error 2
make[1]: Leaving directory '/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc12/external/geant4/11.2.2-2f8e5d3e76b0f6b6f530aab595a96164/build'
make: *** [Makefile:156: all] Error 2
error: Bad exit status from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/rpm-tmp.DlYB4F (%build)


RPM build errors:
line 37: It's not recommended to have unversioned Obsoletes: Obsoletes: external+geant4+11.2.2-2f8e5d3e76b0f6b6f530aab595a96164
Bad exit status from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/rpm-tmp.DlYB4F (%build)
</pre>

--- a/source/geometry/magneticfield/include/G4TMagFieldEquation.hh
+++ b/source/geometry/magneticfield/include/G4TMagFieldEquation.hh
@@ -101,3 +101,4 @@ private:
   T_Field *itsField;
 };
 
+#endif

--- a/source/processes/hadronic/models/de_excitation/multifragmentation/src/G4StatMFMicroPartition.cc
+++ b/source/processes/hadronic/models/de_excitation/multifragmentation/src/G4StatMFMicroPartition.cc
@@ -201,7 +201,7 @@ G4double G4StatMFMicroPartition::CalcPartitionTemperature(G4double U,
 	break;
       }
     }
-    if (!yes) { returmn -1.0; }
+    if (!yes) { return -1.0; }
   }
   
   G4double eps = 1.0e-8*std::abs(Ta-Tb);

